haystack/components/converters/jq_json.py

```diff
@@ -0,0 +1,133 @@
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Union, Iterator, Callable
+
+from haystack import Document, component, logging
+from haystack.components.converters.utils import get_bytestream_from_source, normalize_metadata
+from haystack.lazy_imports import LazyImport
+
+import json
+
+with LazyImport("Run 'pip install jq' Github - https://github.com/jqlang/jq ") as jq_import:
+    import jq
+
+logger = logging.getLogger(__name__)
+
+
+@component
+class JQToJSON:
+    '''
+        :param source:Json source - Can be a string file path or Path object to JSON source
+        :param jq_data_schema: jq query to operate on JSON source
+        :param metadata_func: Optional function that takes in the JSON
+        object extracted by the jq_schema and returns
+        a dict of metadata.  The metadata can be attached to each document
+        that is generated by the JSON source
+        :param metadata:Optional metadata common to all the Documents.
+        This value is a dictionary that is added to all the documents
+        generated by a JSON source.
+        :returns: Dictionary {"documents": List of documents that are the result of jq query applied
+        to the JSON source}
+
+        """
+
+    Example :
+
+        file_path = Path('sample.json')
+        jq_data_schema = '.prizes[].laureates[]?'
+        output_dict =  json_to_doc.run(sources=file_path, jq_data_schema=jq_data_schema,
+        metadata_func=metadata_func, metadata = {'dataset': 'nobel data set'},json_lines = json_lines)
+        documents = output["documents"]
+
+    '''
+
+    def __init__(self: "JQToJSON") -> None:
+        """
+        Check to see if jq is installed
+        """
+        jq_import.check()
+
+    @component.output_types(documents=Dict[str,List[Document]])
```

Comment:
Why `Dict[str,List[Document]]` over just `List[Document]`?

Reply:
Either is fine with me.  If Dict[str,List[Document]] is confusing we can leave it at List[Document] as long as we implicitly understand that a dictionary is being returned.

Reply:
Removed